src/java/org/apache/cassandra/db/Memtable.java

```diff
@@ -68,22 +70,27 @@
 
     private static MemtablePool createMemtableAllocatorPool()
     {
-        long heapLimit = DatabaseDescriptor.getMemtableHeapSpaceInMb() << 20;
-        long offHeapLimit = DatabaseDescriptor.getMemtableOffheapSpaceInMb() << 20;
+        final long heapLimit = DatabaseDescriptor.getMemtableHeapSpaceInMb() << 20;
+        final long offHeapLimit = DatabaseDescriptor.getMemtableOffheapSpaceInMb() << 20;
+        final float cleaningThreshold = DatabaseDescriptor.getMemtableCleanupThreshold();
+        final MemtableCleaner cleaner = ColumnFamilyStore::flushLargestColumnFamily;
+        final int maxPendingTasks = Integer.getInteger(Config.PROPERTY_PREFIX + "max_pending_flushing_tasks", ColumnFamilyStore.getNumFlushWriters() * 2);
+
         switch (DatabaseDescriptor.getMemtableAllocationType())
```

Comment:
Why don't we just attach a lambda with the proper creation logic to each of the allocation types in the enum? The arguments are basically the same, with the off-heap limit being ignored in one case. It would make this whole thing one line, avoid the silly `AssertionException`, and maybe even make the new local variables we have here unnecessary even for style purposes.